pkg/util/probe.go

```diff
@@ -0,0 +1,104 @@
+package util
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+/*
+Common functionality to add readiness/liveness Probe to a container using:
+
+* default values:
+readinessProbe 	:= common.Probe{ProbeType: "readiness"}
+livenessProbe 	:= common.Probe{ProbeType: "liveness"}
+
+* custom values:
+readinessProbe := common.Probe{
+	ProbeType: 		"readiness",
+	Command:		"/to/some/command",
+	InitialDelaySeconds: 	20,
+ }
+livenessProbe := common.Probe{ProbeType: "liveness"}
+
+In container definition
+Containers: []corev1.Container{
+	{
+		Name:  "nova-conductor",
+		ReadinessProbe: readinessProbe.GetProbe(),
+		LivenessProbe:  livenessProbe.GetProbe(),
+		...
+*/
+
+const (
+	// default Readiness values
+	defaultReadinessInitialDelaySeconds int32 = 5
+	defaultReadinessPeriodSeconds       int32 = 15
+	defaultReadinessTimeoutSeconds      int32 = 3
+	defaultReadinessFailureThreshold    int32 = 3
+	// default Liveness values
+	defaultLivenessInitialDelaySeconds int32  = 30
+	defaultLivenessPeriodSeconds       int32  = 60
+	defaultLivenessTimeoutSeconds      int32  = 3
+	defaultLivenessFailureThreshold    int32  = 5
+	defaultCommand                     string = "/openstack/healthcheck"
+)
+
+// Probe details
+type Probe struct {
+	// ProbeType, either readiness, or liveness
+	ProbeType           string
```

Comment:
Define a ProbeType type here and use that instead of string. Define "readiness" and "liveness" as constants of this type.

Reply:
donw